sonar-plugin-api/src/main/java/org/sonar/api/server/rule/RulesDefinition.java

```diff
@@ -1077,6 +1092,55 @@ public Param param(String key) {
       return tags;
     }
 
+    /**
+     * Deprecated rules keys for this rule.
+     * <p>
+     * If you want to rename the key of a rule or change its repository or both, register the rule's previous repository
+     * and key.
+     * <br>
+     * This will allow SonarQube to support "issue renaming" (ie. <strong>not</strong> closing existing issues
+     * for this rules, created under the rule's previous repository and/or name, and recreate them as new under the
```

Comment:
@sns-seb 
> recreate them as new

they will not appear as new issues from user point of view, right? If right, we probably, should rephrase it

Reply:
it seems it not enough, I will rephrase